src/bootstrap/src/core/config/toml/mod.rs

```diff
@@ -152,10 +152,6 @@ impl Config {
     }
 
     pub(crate) fn get_toml(file: &Path) -> Result<TomlConfig, toml::de::Error> {
-        #[cfg(test)]
-        return Ok(TomlConfig::default());
-
-        #[cfg(not(test))]
         Self::get_toml_inner(file)
```

Comment:
Just to make sure that we didn't miss anything, and that we won't misuse this in the future, could you please add a check where if we're in `#[cfg(test)]`, we assert that `file` does not come from the source directory? We can find that out from `CARGO_MANIFEST_DIR`'s grandparent. To avoid a situation where a test would mistakenly read `<src>/bootstrap.toml`.